lib/rex/post/meterpreter/ui/console/command_dispatcher/priv/passwd.rb

```diff
@@ -39,14 +39,49 @@ def name
   def cmd_hashdump(*args)
     client.priv.sam_hashes.each { |user|
       print_line("#{user}")
+      if shell.client.platform == 'windows' && !shell.framework.nil? && shell.framework.db.active
+        report_creds(user)
+      end
     }
 
     return true
   end
 
-end
+  def report_creds(user_data)
+    user = user_data.user_name
+    lm_hash = user_data.lanman.downcase
+    nt_hash = user_data.ntlm.downcase
+    empty_password = lm_hash.eql?('aad3b435b51404eeaad3b435b51404ee') && nt_hash.eql?('31d6cfe0d16ae931b73c59d7e0c089c0')
```

Comment:
For context, cause I was also confused about this, https://rubystyle.guide/#eql recommends using == over `.eql?` where the extra features of `.eql?` are not useful. Furthermore as pointed out in a conversation with @smcintyre-r7, the `lib/` directory alone has 2819 uses of `==` and only 36 usages of `.eql?`.

Whilst I agree from a visual perspective that `.eql?` may look nicer, its better if we keep to standards in the library code rather than having different pieces of code use different standards, particularly for library code that will be frequently reused.

I'll be updating this in the next commit.